packages/react-union-scripts/scripts/lib/utils.js

```diff
@@ -202,26 +207,14 @@ const getUnionConfig = () =>
 const getAppConfig = () =>
 	R.find(R.whereEq({ name: isMonoRepo ? cli.appOriginal : cli.app }), getUnionConfig());
 
-const resolveAsyncSuffix = R.cond([
-	[R.is(RegExp), R.identity],
-	[R_.isString, asyncSuffix => R_.constructRegExp(`\.${asyncSuffix}\.js$`, 'i')],
-	[R_.isArray, asyncSuffix => R_.constructRegExp(`\.(${R.join('|', asyncSuffix)})\.js$`, 'i')],
-	[
-		R.T,
-		() =>
-			invariant(false, "Invalid property 'asyncSuffix'. It should be string or list of strings."),
-	],
-]);
-
-const mergeWhen = (condition, fn, ...fnArgs) => (condition ? fn(...fnArgs) : {});
+const mergeWhen = (condition, fn, args = []) => (condition ? fn(...args) : {});
```

Comment:
I can revert this change if you want, but it was driving me nuts 😄 I absolutely hate this function.

Reply:
You can leave it if you want. But it just uses `Function.prototype.apply()` syntax instead of `Function.prototype.call()`.

Reply:
It's true, but when I was looking at the usage of the function, I had no clue what it was supposed to do, the arguments are just sort of scattered there. I like variadic functions if no other parameters are present, but having a variadic function with 2 named parameters... I don't like that.